### ISSUE TITLE
Protected Routes with Preact Router

### ISSUE THREAD

**Issue by nishtacular** [developit/preact-router#55]
Hey @developit, 

I'm running into some issues trying to implement protected routes. I know react-router has the onEnter prop which helps implement protected routes basically admin pages, etc. How would one implemented this with preact-router. 

This is my current approach

```
// Excerpt from Root Container
render (props) {
  return (
    <Router>
      <Home path='/' user={props.user}/>
      <Contact path='/contact' />
      <ProfileContainer path='/profile' />
      <LoginContainer path='/signin' />
      <PageError type='404' default />
    </Router>
  )
}
```
Then inside LoginContainer I'm checking if the session is valid, but componentWillMount is not being triggered.
```
...
componentWillMount () {
  if (isLoggedIn) {
    route('/profile')
  }
}

render (props) {
...
```

Not sure if this is the right approach. Any pointers will be greatly appreciated. Thanks

**Comment by developit** [comment]
Hi @nishtacular - you're exactly right, just there's currently a tiny bug in `preact-router` for this exact use-case.  For now, please wrap your call to `route()` in a setTimeout:

```js
...
componentWillMount () {
  if (isLoggedIn) {
    setTimeout( () => route('/profile') );
  }
}

render (props) {
...
```

If you want to avoid the few millisecond flash of your view, perform the same `isLoggedIn` check in `render()`.  

Hopefully I'll land a fix for this soon!

**Comment by developit** [comment]
If you don't mind, I'm going to close this out as a duplicate since we're already tracking the same fix in #48. Cheers!
